src/main/java/usi/si/seart/repository/specification/GitRepoSearch.java

```diff
@@ -0,0 +1,176 @@
+package usi.si.seart.repository.specification;
+
+import com.google.common.collect.Range;
+import lombok.Builder;
+import org.apache.commons.lang3.StringUtils;
+import usi.si.seart.model.GitRepo;
+import usi.si.seart.model.GitRepo_;
+import usi.si.seart.model.GitRepoLabel_;
+import usi.si.seart.model.GitRepoMetric_;
+import usi.si.seart.model.MetricLanguage_;
+import usi.si.seart.repository.criteria.Criteria;
+import usi.si.seart.repository.criteria.KeyCriteria;
+import usi.si.seart.repository.criteria.KeyValueCriteria;
+import usi.si.seart.repository.operation.BinaryOperation;
+import usi.si.seart.repository.operation.UnaryOperation;
+
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Root;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Builder
+public class GitRepoSearch {
+
+    String name;
+    boolean nameEquals;
+    String language;
+    String license;
+    String label;
+    Range<Long> commits;
+    Range<Long> contributors;
+    Range<Long> issues;
+    Range<Long> pulls;
+    Range<Long> branches;
+    Range<Long> releases;
+    Range<Long> stars;
+    Range<Long> watchers;
+    Range<Long> forks;
+    Range<Date> created;
+    Range<Date> committed;
+    Range<Long> codeLines;
+    Range<Long> commentLines;
+    Range<Long> totalLines;
+    boolean hasCodeMetricsFilters;
+    boolean excludeForks;
+    boolean onlyForks;
+    boolean hasIssues;
+    boolean hasPulls;
+    boolean hasWiki;
+    boolean hasLicense;
+
+    @SuppressWarnings("unchecked")
```

Comment:
This was the primary reason I wanted to migrate from just using a `Map` to a dedicated object: we can completely forgo unchecked casts. With the changes you made, this suppression is no longer needed ;)